### ISSUE TITLE
Feed bayesian optimizer with previous experiment's results

### ISSUE THREAD

**Issue by jukiewiczm** [guildai/guildai#115]
I'm trying to do some hyperparameter optimization using the bayesan optimizer. The problem is, I cannot run it all in one go and need to terminate the process from time to time.

Every time I do this and start the experiment again, I see:
```
INFO: [guild] Found 0 previous trial(s) for use in optimization
```
That's no good, because it means every time i re-run the experiment, I'm basically starting over.

Is there a way to run the optimization in a way that would feed all the previous runs to the optimizer before running? That would be quite useful.

**Comment by gar1t** [comment]
Guild doesn't look outside the batch run for trials because the results may be not be comparable for various reasons (e.g. code changes, different data sets, etc.)

You can restart an optimization run using the `--restart` option. The optimization run is what controls the trial generation process. These runs are identified by the `+` in the operation names. E.g. if you run `train` with the `gp` optimizer, you'll see a run with the operation name `train+gp`.

Find the run ID for the optimizer run and restart it:

    $ guild run --restart <run ID>

You can change flag search ranges, set a different `--max-trials` val, etc.

One thing to bear in mind, the optimizers are configured to use three random starts (at least in the current 0.7 pre-release, which I recommend that you use). When you restart the run, it will use three more random starts. You can change this this way:

    $ guild run --restart <run ID> -Fo random-starts=0

The `-Fo` is how you set an optimizer flag.

**Comment by jukiewiczm** [comment]
Thanks Gareth, that's helpful! I assume this works so I'm closing the task (but will reopen if turns out otherwise).
Would love to see that documented though :)).

**Comment by jukiewiczm** [comment]
@gar1t What's the difference between --restart and --rerun? It doesn't seem clear from the --help. Didn't you mean --rerun by the way?

**Comment by gar1t** [comment]
You want `--restart` for your case, not `--rerun`.

`--rerun` starts a new run using the settings from the specified run - so you get a second run. `--restart` starts the specified run in place, using the original run directory.

**Comment by jukiewiczm** [comment]
Hey @gar1t , one more question regarding that.

I'm running the python file straight away (like `guild run xxx.py ... --optimizer gp --max-trials 100`)
And after several runs (>3) I can't really see any `xxx+gp` label.

Do I need to create `guild.yml` and define operation there for this solution to work?

**Comment by gar1t** [comment]
When you say you can't see any xxx+gp label, are you saying the batch run no longer shows up in the runs list?

Guild only shows the latest 20 runs by default - if you want to show more, use `-m` (for another 20, can use multiple times) or `-a` to show all.

Or you can can filter by operation name using `-o` - e.g. `guild runs -o xxx.py+gp`

Using a Guild file here should not effect any of this behavior.

**Comment by jukiewiczm** [comment]
Oh, I just noticed what I was doing wrong.
I was looking for the `xxx+gp` in the `guild compare` command, not the `guild runs` command.

I should be fine now :). Will let you know if that worked. Thanks!

EDIT: After finding the proper run using `guild runs` and restarting the `xxx+gp` run, it worked!
